File: business/src/main/kotlin/com/isystk/sample/domain/entity/OrderHistory.java
package com.isystk.sample.domain.entity;

import com.isystk.sample.domain.dto.DomaDtoImpl;
import java.math.BigInteger;
import java.time.LocalDateTime;
import org.seasar.doma.Column;
import org.seasar.doma.Entity;
import org.seasar.doma.GeneratedValue;
import org.seasar.doma.GenerationType;
import org.seasar.doma.Id;
import org.seasar.doma.Table;
import org.seasar.doma.Version;

/**
 * 注文履歴
 *
 * 自動生成のため原則修正禁止!!
 *
 */
@Entity
@Table(name = "order_history")
public class OrderHistory extends DomaDtoImpl {

    /** serialVersionUID */
    private static final long serialVersionUID = 1L;


    /** 注文履歴ID */
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    BigInteger id;

    /** 商品ID */
    @Column(name = "stock_id")
    BigInteger stockId;

    /** ユーザID */
    @Column(name = "user_id")
    BigInteger userId;

    /** 価格 */
    @Column(name = "price")
    Integer price;

    /** 個数 */
    @Column(name = "quantity")
    Integer quantity;

    /** 登録日時 */
    @Column(name = "created_at")
    LocalDateTime createdAt;

    /** 更新日時 */
    @Column(name = "updated_at")
    LocalDateTime updatedAt;

    /** 削除フラグ */
    @Column(name = "delete_flg")
    Boolean deleteFlg;

    /** 楽観チェック用バージョン */
    @Version
    @Column(name = "version")
    Long version;

    public BigInteger getId() {
        return this.id;
    }

    public BigInteger getStockId() {
        return this.stockId;
    }

    public BigInteger getUserId() {
        return this.userId;
    }

    public Integer getPrice() {
        return this.price;
    }

    public Integer getQuantity() {
        return this.quantity;
    }

    public LocalDateTime getCreatedAt() {
        return this.createdAt;
    }

    public LocalDateTime getUpdatedAt() {
        return this.updatedAt;
    }

    public Boolean getDeleteFlg() {
        return this.deleteFlg;
    }

    public Long getVersion() {
        return this.version;
    }

    public void setId(BigInteger id) {
        this.id = id;
    }

    public void setStockId(BigInteger stockId) {
        this.stockId = stockId;
    }

    public void setUserId(BigInteger userId) {
        this.userId = userId;
    }

    public void setPrice(Integer price) {
        this.price = price;
    }

    public void setQuantity(Integer quantity) {
        this.quantity = quantity;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public void setUpdatedAt(LocalDateTime updatedAt) {
        this.updatedAt = updatedAt;
    }

    public void setDeleteFlg(Boolean deleteFlg) {
        this.deleteFlg = deleteFlg;
    }

    public void setVersion(Long version) {
        this.version = version;
    }
}

File: business/src/main/kotlin/com/isystk/sample/domain/entity/OrderHistory.kt
package com.isystk.sample.domain.entity;

import com.isystk.sample.domain.dto.DomaDtoImpl
import org.seasar.doma.*
import java.time.LocalDateTime

/**
 * 注文履歴
 *
 * 自動生成のため原則修正禁止!!
 *
 */
@Entity
@Table(name = "order_history")
open class OrderHistory : DomaDtoImpl() {


    /** 注文履歴ID */
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    var id: Long? = null

    /** 商品ID */
    @Column(name = "stock_id")
    var stockId: Long? = null

    /** ユーザID */
    @Column(name = "user_id")
    var userId: Long? = null

    /** 価格 */
    @Column(name = "price")
    var price: Long? = null

    /** 個数 */
    @Column(name = "quantity")
    var quantity: Long? = null

    /** 登録日時 */
    @Column(name = "created_at")
    var createdAt: LocalDateTime? = null

    /** 更新日時 */
    @Column(name = "updated_at")
    var updatedAt: LocalDateTime? = null

    /** 削除フラグ */
    @Column(name = "delete_flg")
    var deleteFlg: Boolean? = null

    /** 楽観チェック用バージョン */
    @Version
    @Column(name = "version")
    var version: Long? = null

    companion object {
        /** serialVersionUID  */
        private const val serialVersionUID = 1L
    }
}